modules/actions/user/upsert-bpt-balances-v3.ts

```diff
@@ -0,0 +1,33 @@
+import { Chain } from '@prisma/client';
+import { V3VaultSubgraphClient } from '../../sources/subgraphs';
+import { prisma } from '../../../prisma/prisma-client';
+import { poolShareToUserBalance } from '../../sources/transformers/pool-share-to-user-balance';
+
+export const upsertBptBalancesV3 = async (vaultSubgraphClient: V3VaultSubgraphClient, chain: Chain) => {
+    const poolShares = await vaultSubgraphClient.getAllPoolShares();
```

Comment:
this should be retrieved onchain with events. Using the subgraph only to init the shares as we want as up to date shares as possible.

Reply:
onchain is already covered in the current "syncChangedBalancesForAllPools"